### ISSUE TITLE
Track actual storage

### ISSUE THREAD

**Issue by shawkins** [strimzi/strimzi-kafka-operator#5967]
**Is your feature request related to a problem? Please describe.**
Strimzi provides the configured storage in the Kafka StatefulSet and Pod annotations, but it doesn't show what the actual available capacity is.  In situations where a resize has been requested that may be in a pending state for quite a while - for example on AWS with EBS you can only resize a pvc once every 6 hours.

**Describe the solution you'd like**
Add a status field showing the actual amount of storage capacity.

**Comment by scholzj** [comment]
The annotations are for internal use only. Strimzi does not really track the progress of resizing and AFAIK it does not need to. The resizing process is controlled by Kubernetes and you should look at the PVCs / PVs for the resizing progress, eventually monitor the Kubernetes logs which would normally inform you that the resizing is not possible due tot he limits.

**Comment by shawkins** [comment]
> Strimzi does not really track the progress of resizing and AFAIK it does not need to.  The resizing process is controlled by Kubernetes and you should look at the PVCs / PVs for the resizing progress, eventually monitor the Kubernetes logs which would normally inform you that the resizing is not possible due tot he limits.

Yes we are looking to add an Informer on PVCs and update our status based upon their status storage capacity, but it seemed like a potential task for strimzi given that the kafka resource is the owner of the pvcs.  I'll close this based upon your feedback.

**Comment by scholzj** [comment]
> I'll close this based upon your feedback.

Up to you I guess, you can always keep it open for others to comment as well or you can come for the community call to discuss it ;-).

One thing I learned is that this area is badly standardised, so one thing to consider would be how does it work with different environments / providers etc. If you are interested just in AWS, it might make it easier to have some custom solution designed for that. Because if we do it in Strimzi, we would need to be searching for the common denominators.

**Comment by shawkins** [comment]
> Up to you I guess, you can always keep it open for others to comment as well or you can come for the community call to discuss it ;-).

I interpreted your response to mean strimzi doesn't need to do this.  If you are fine with leaving the issue open that's not a problem to me.

> Because if we do it in Strimzi, we would need to be searching for the common denominators.

The long resizing cycle takes place on AWS, but the PVC.status.capacity.storage should be reported the same regardless. 

What we are looking to do is have a single status value that conveys the currently allocated storage in addition to the requested storage.  Strimzi already adds a warning condition when attempting to resize to a smaller size, so there is some consideration of a discrepancy in that scenario.

**Comment by scholzj** [comment]
> I interpreted your response to mean strimzi doesn't need to do this. If you are fine with leaving the issue open that's not a problem to me.

Well, that is indeed my opinion. But I'm not Strimzi ... so all it is is an opinion of one maintainer of many which is really the point I was trying to make ;-)

> The long resizing cycle takes place on AWS, but the PVC.status.capacity.storage should be reported the same regardless.

> What we are looking to do is have a single status value that conveys the currently allocated storage in addition to the requested storage. Strimzi already adds a warning condition when attempting to resize to a smaller size, so there is some consideration of a discrepancy in that scenario.

We do validation of the storage changes to actually avoid doing to much monitoring of the PVC and PV statuses. So we validate it and reject it in our code (this is what the annotations are for really). But if the change is valid, we do not do much => we just update the PVC and that is it.

All the time - regardless whether any resizing was requested or not - we check for some selected signals from the PVCs that a restart of the pods is needed to resize the filesystem. But this is not dependent on any resized state since in some cases, the resizing is not done through Strimzi but directly on the PVs / PVCs (as I said, little standardisation around this). So we do not really want to track the resizing state for this.

So basically, there is no need to track the resizing progress for us.

So the question I guess is what value is in Strimzi doing this? Aggregating the data from PVCs and adding it to another resource doesn't seem to add much value. The total sum of storage is anyway not interesting with Kafka because it does not tell anything about when you run out of storage etc. The available capacity / used capacity is part of the Kube metrics so there does not seem to be some new value either.

**Comment by scholzj** [comment]
Triaged on 7.6.2022: Constantly updating the Kafka CR just to update the storage usage would create a lot of noise in the CO logs. The informatioon is already available in different ways:
* Other Kubernetes resources
* Kubernetes metrics
* Kafka metrics offer some infromation about used storage
* In the future, [KIP-827](https://cwiki.apache.org/confluence/display/KAFKA/KIP-827%3A+Expose+logdirs+total+and+usable+space+via+Kafka+API) might add an API to Kafka Admin API
